### ISSUE TITLE
Use NavigationSplitView instead of NavigationView

### ISSUE THREAD

**Issue by Audiotape-2** [migueldeicaza/SwiftTermApp#205]
Hi Miguel, 

since the NavigationView is [already deprecated](https://developer.apple.com/documentation/swiftui/navigationview) I was wondering how one would use the ConfigurableUITerminal View in conjunction with a NavigationSplitView. 

The problem I'm running into is that the first time I select a Host terminal shows up, everything is working fine. 
But as soon as I select a different host, nothing changes and there's still the first host's terminal displayed. 
Is it even possible to use a NavigationSplitView without major refactoring? 

All the best, 
Johannes

**Comment by migueldeicaza** [comment]
It is a messy dance involving the “rehosting” feature.  It is not really the navigation stack problem (it doesn’t help, but it isn’t the source of the problems)

The problem is really that this code is doing some ugly stuff to show the same view in multiple places when it shouldn’t.

a long term fix is to make TerminalView a pure view for the Terminal controller in SwiftTerm or avoid doing this dance

LaTerminalApp has the dance upgraded, but it has been so many changes that it is hard to extract a fix

**Comment by migueldeicaza** [comment]
Yes, there is an issue filed in SwiftTerm, and I have a design doc I used a month or so ago when I first tried it
